content/articles/How-to-start-up-with-Socket-Programming-with-Java/index.md

```diff
@@ -2,22 +2,22 @@
 This is an article about Sockets. It will give you a glimpse of how clients and servers communicate over the internet.
 
 Table of content:
-- [What is a socket](#what-is-a-socket)
-- [Programming on Server-side](#programming-on-server-side)
-- [](#)
-- [](#)
-- [](#)
+- [socket definition](#socket-definition)
+- [Server-Side Programming](#server-side-programming)
+- [Client-Side Programming](#client-side-programming)
+- [How Run The Programs on different applications](#how-to-run-the-programs-on-diffrent-applications)
+- [Conclusion](#conclusion)
 
 ### Introduction
-Socket programming is a way of connecting two computers on a network to allow the exchange of information between the two machines that are running over `JRE`(Java Runtime Environment). The connection can be through **connection-oriented protocol** or **connectionless** but for our case, we will be using connection-oriented which is TCP/IP.
+Socket programming is a way of connecting two computers over a network to allow the exchange of information.The connection can be through **connection-oriented protocol** or **connectionless**. In our case, we will be using  TCP/IP a connection-oriented protocol.
 
-TCP is a connection-oriented protocol, which means that before data can be transferred, two computers must first establish a connection.
+In connection-oriented protocols, two computers must establish a connection before transferring data.
 
-Connectionless -For this type of connection, UDP protocol (User Datagram Protocol)also a layer 2 protocol, is used to provide a good picture where you do not necessarily need every packet to arrive at its desired destination.
+Connectionless -For this type of connection, UDP protocol (User Datagram Protocol)also a layer 2 protocol, it is used where you do not necessarily need every packet to arrive at its desired destination.
 
-We will be using Client\Server architecture to illustrate more on sockets. Client and server communicate by writing to and reading from socket connection thus for a connection to happen between the server and a client, a socket must be present. The two machines should know each other's network information.
+We will be using Client\Server architecture to illustrate more on sockets.The Client and server communicate by writing to and reading from socket connection .To establish this connection,the machines must have each other's network information.
 
-### What is a Socket?
+### Socket definition
 A socket is an **endpoint** of communication, acting as a connection between two computers on a network. It contains a socket number that the TCP layer can use to identify the application to which data will be transmitted. An endpoint is made up of a port number and an IP address.
```

Comment:
Change: 

> An endpoint is made up of a port number and an IP address.

To:

**Additionally, an endpoint is made up of a port number and an IP address.**

Just adding **Additionally,** here as a transition word to make this sentence fit in more. Otherwise, nice job.